src/scripts/domParsers/domUtils.ts

```diff
@@ -2,179 +2,190 @@
 // they rename `sanitize` to `sanitizeHtml`, which is the name of the actually exported function
 declare let sanitizeHtml;
 
+// This private has technically been deprecated, but is present in all major browsers as of 2/12/14 and offers
+// the best perf for the UTF byte manipulation we need for truncation and byte size estimation.
+// DO NOT USE IT WITHOUT CHECKING IF IT EXISTS -- in case browser vendors actually remove them
+// ReSharper disable once InconsistentNaming
+declare function unescape(s: string): string;
+
 import {Constants} from "../constants";
 import {ObjectUtils} from "../objectUtils";
-import {VideoUtils} from "./videoUtils";
+import {SupportedVideoDomains, VideoUtils} from "./videoUtils";
 
 import {VideoExtractorFactory} from "./VideoExtractorFactory";
 
+export interface EmbeddedVideoIFrameSrcs {
+	srcAttribute: string;
+	dataOriginalSrcAttribute: string;
+}
+
 /**
  * Dom specific Helper utility methods
  */
-export module DomUtils {
-	export module Tags {
-		export const a = "a";
-		export const b = "b";
-		export const applet = "applet";
-		export const article = "article";
-		export const audio = "audio";
-		export const base = "base";
-		export const body = "body";
-		export const br = "br";
-		export const button = "button";
-		export const canvas = "canvas";
-		export const center = "center";
-		export const cite = "cite";
-		export const del = "del";
-		export const div = "div";
-		export const em = "em";
-		export const embed = "embed";
-		export const figure = "figure";
-		export const font = "font";
-		export const h1 = "h1";
-		export const h2 = "h2";
-		export const h3 = "h3";
-		export const h4 = "h4";
-		export const h5 = "h5";
-		export const h6 = "h6";
-		export const head = "head";
-		export const header = "header";
-		export const hr = "hr";
-		export const html = "html";
-		export const i = "i";
-		export const iframe = "iframe";
-		export const img = "img";
-		export const input = "input";
-		export const li = "li";
-		export const link = "link";
-		export const main = "main";
-		export const map = "map";
-		export const menu = "menu";
-		export const menuitem = "menuitem";
-		export const meta = "meta";
-		export const meter = "meter";
-		export const noscript = "noscript";
-		export const object = "object";
-		export const ol = "ol";
-		export const p = "p";
-		export const progress = "progress";
-		export const script = "script";
-		export const span = "span";
-		export const source = "source";
-		export const strike = "strike";
-		export const strong = "strong";
-		export const style = "style";
-		export const sub = "sub";
-		export const sup = "sup";
-		export const svg = "svg";
-		export const table = "table";
-		export const td = "td";
-		export const title = "title";
-		export const tr = "tr";
-		export const u = "u";
-		export const ul = "ul";
-		export const video = "video";
-	}
+export class DomUtils {
+	public static tags = {
```

Comment:
Side thought: I'm wondering if we should stop using these constants and use hardcoded strings instead. I think constants aren't very useful on immutable values, and it's much more readable if we use hardcoded strings (i.e., the javascript way of doing things).